core/payment/src/lib.rs

```diff
@@ -53,8 +55,10 @@ impl PaymentService {
         let db = context.component();
         db.apply_migration(migrations::run_with_output)?;
 
+        let config = Config::from_env()?;
+
         let processor = Arc::new(PaymentProcessor::new(db.clone()));
-        self::service::bind_service(&db, processor.clone(), BindOptions::default());
+        self::service::bind_service(&db, processor.clone(), BindOptions::default(), config);
```

Comment:
I would pass Arc<Config> down the hierarchy.
Since Config struct is container for all parameters of payments module, than you must be able to clone the struct cheaply.

Reply:
Done.